src/vm/amd64/cgenamd64.cpp

```diff
@@ -1126,6 +1126,121 @@ PCODE DynamicHelpers::CreateHelperWithTwoArgs(LoaderAllocator * pAllocator, TADD
     END_DYNAMIC_HELPER_EMIT();
 }
 
+//
+// Move an an argument into the second argument register and jump to a helper function.
+// Assumption: the first argument is already placed into the first argument register.
+//
+void DynamicHelpers::EmitJumpStubWithSecondArg(LoaderAllocator * pAllocator, BYTE*& pCode, PCODE pTarget, TADDR secondArgValue)
+{
+    CONTRACTL
+    {
+        GC_NOTRIGGER;
+        PRECONDITION(pCode != NULL && pTarget != NULL);
+    }
+    CONTRACTL_END;
+
+    // Move second argument into argument register (size = 10)
+#ifdef UNIX_AMD64_ABI
+    *(UINT16 *)pCode = 0xBE48; // mov rsi, XXXXXX
+#else
+    *(UINT16 *)pCode = 0xBA48; // mov rdx, XXXXXX
+#endif
+    pCode += 2;
+    *(TADDR *)pCode = secondArgValue;
+    pCode += 8;
+
+    // Jump stub (size = 5)
+    *pCode++ = X86_INSTR_JMP_REL32; // jmp rel32
+    *(INT32 *)pCode = rel32UsingJumpStub((INT32 *)pCode, pTarget, NULL, pAllocator);
+    pCode += 4;
+}
+
+PCODE DynamicHelpers::CreateDictionaryLookupHelper(LoaderAllocator * pAllocator, CORINFO_RUNTIME_LOOKUP * pLookup)
+{
+    STANDARD_VM_CONTRACT;
+
+    // TODO: fix codegen for correct Unix ABI...
+
+    // It's available only via the run-time helper function
+    if (pLookup->indirections == CORINFO_USEHELPER)
+    {
+        BEGIN_DYNAMIC_HELPER_EMIT(16);
+
+        // rcx contains the generic context parameter
+        // mov rdx,pLookup->signature
+        // call pLookup->helper
+        EmitJumpStubWithSecondArg(pAllocator, p, CEEJitInfo::getHelperFtnStatic(pLookup->helper), (TADDR)pLookup->signature); // 15 bytes
+
+        // ret
+        *p++ = 0xC3;
+
+        END_DYNAMIC_HELPER_EMIT();
+    }
+    else
+    {
+        int codeSize = (pLookup->indirections * 4) + (pLookup->testForNull ? 28 : 4);
+
+        BEGIN_DYNAMIC_HELPER_EMIT(codeSize);
+
+        if (pLookup->testForNull)
+        {
+            // rcx contains the generic context parameter. Push on the stack to save a copy of it.
+            *p++ = 0x51;    // push rcx
```

Comment:
It would be better to mov into scratch register like rax instead of pushing to stack.